templates/stake-pool-template/move/sources/stake_pool.move

```diff
@@ -0,0 +1,519 @@
+            reward_store: fungible_asset::create_store(
+                reward_store_constructor_ref,
+                object::address_to_object<Metadata>(@reward_fa_obj_addr)
+            ),
+            user_stakes: table::new(),
+            total_stake: 0,
+            reward_schedule: option::none(),
+        });
+    }
+
+    // ================================= Entry Functions ================================= //
+
+    /// Set pending admin of the contract, then pending admin can call accept_admin to become admin
+    public entry fun set_pending_admin(sender: &signer, new_admin: address) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@stake_pool_addr);
+        assert!(is_admin(config, sender_addr), ERR_ONLY_ADMIN_CAN_SET_PENDING_ADMIN);
+        config.pending_admin = option::some(new_admin);
+    }
+
+    /// Accept admin of the contract
+    public entry fun accept_admin(sender: &signer) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@stake_pool_addr);
+        assert!(config.pending_admin == option::some(sender_addr), ERR_ONLY_PENDING_ADMIN_CAN_ACCEPT_ADMIN);
+        config.admin = sender_addr;
+        config.pending_admin = option::none();
+    }
+
+    public entry fun create_reward_schedule(
+        sender: &signer,
+        rps: u64,
+        duration_seconds: u64
+    ) acquires StakePool, Config {
+        let current_ts = timestamp::now_seconds();
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global<Config>(@stake_pool_addr);
+        assert!(config.reward_creator == sender_addr, ERR_ONLY_REWARD_CREATOR_CAN_ADD_REWARD);
+
+        let total_reward_amount  = rps * duration_seconds;
+        let stake_pool_mut = borrow_global_mut<StakePool>(@stake_pool_addr);
+        assert!(option::is_none(&stake_pool_mut.reward_schedule), ERR_REWARD_SCHEDULE_ALREADY_EXISTS);
+        assert!(primary_fungible_store::balance(sender_addr, stake_pool_mut.reward_fa_metadata_object) >= total_reward_amount, ERR_NOT_ENOUGH_BALANCE_TO_ADD_REWARD);
+
+        stake_pool_mut.reward_schedule = option::some(RewardSchedule {
+            index: fixed_point64::create_from_u128(0),
+            rps,
+            last_update_ts: current_ts,
+            start_ts: current_ts,
+            end_ts: current_ts + duration_seconds,
+        });
+
+        fungible_asset::transfer(
+            sender,
+            primary_fungible_store::primary_store(sender_addr, stake_pool_mut.reward_fa_metadata_object),
+            stake_pool_mut.reward_store,
+            total_reward_amount,
+        );
+    }
+
+    public entry fun claim_reward(sender: &signer) acquires StakePool, RewardStoreController {
+        let current_ts = timestamp::now_seconds();
+        let sender_addr = signer::address_of(sender);
+        let stake_pool = borrow_global<StakePool>(@stake_pool_addr);
```

Comment:
do we need for an assert the sender has rewards to claim?

Reply:
if sender doesn't have reward to claim, this function won't do anything. or you are saying it's better to error out?

Reply:
user can call `get_claimable_reward` first, if it's not 0 then call claim